x-pack/plugins/cases/public/components/custom_fields/list/edit.tsx

```diff
@@ -0,0 +1,278 @@
+interface FormState {
+  value: string;
+  isValid: boolean | undefined;
+  submit: FormHook<{ value: string }>['submit'];
+}
+
+interface FormWrapper {
+  initialValue: string;
+  isLoading: boolean;
+  customFieldConfiguration: ListCustomFieldConfiguration;
+  onChange: (state: FormState) => void;
+}
+
+const FormWrapperComponent: React.FC<FormWrapper> = ({
+  initialValue,
+  customFieldConfiguration,
+  isLoading,
+  onChange,
+}) => {
+  const { form } = useForm<{ value: string }>({
+    defaultValue: {
+      value:
+        customFieldConfiguration?.defaultValue != null && isEmpty(initialValue)
+          ? customFieldConfiguration.defaultValue
+          : initialValue,
+    },
+  });
+  const [{ value }] = useFormData({ form });
+  const { submit, isValid } = form;
+  const formFieldConfig = getListFieldConfig({
+    required: customFieldConfiguration.required,
+    label: customFieldConfiguration.label,
+  });
+  const populatedWithDefault =
+    value === customFieldConfiguration?.defaultValue && isEmpty(initialValue);
+
+  useEffect(() => {
+    onChange({
+      value,
+      isValid,
+      submit,
+    });
+  }, [isValid, onChange, submit, value]);
+
+  const selectOptions = useMemo(
+    () => listCustomFieldOptionsToEuiSelectOptions(customFieldConfiguration.options),
+    [customFieldConfiguration.options]
+  );
+
+  return (
+    <Form form={form}>
+      <UseField
+        path="value"
+        config={formFieldConfig}
+        component={SelectField}
+        helpText={populatedWithDefault && POPULATED_WITH_DEFAULT}
+        componentProps={{
+          euiFieldProps: {
+            fullWidth: true,
+            disabled: isLoading,
+            isLoading,
+            'data-test-subj': `case-list-custom-field-form-field-${customFieldConfiguration.key}`,
+            options: selectOptions,
+          },
+        }}
+      />
+    </Form>
+  );
+};
+
+FormWrapperComponent.displayName = 'FormWrapper';
+
+const EditComponent: CustomFieldType<CaseCustomFieldList, ListCustomFieldConfiguration>['Edit'] = ({
+  customField,
+  customFieldConfiguration,
+  onSubmit,
+  isLoading,
+  canUpdate,
+}) => {
+  const selectedKey = customField?.value ? Object.keys(customField.value)[0] : null;
+  const selectedOptionExists = customFieldConfiguration.options.find(
+    (option) => option.key === selectedKey
+  );
+  const initialValue = selectedOptionExists ? selectedKey ?? '' : '';
+
+  const [isEdit, setIsEdit] = useState(false);
+  const [formState, setFormState] = useState<FormState>({
+    isValid: undefined,
+    submit: async () => ({ isValid: false, data: { value: '' } }),
+    value: initialValue,
+  });
+
+  // When updating the field, store the selected value while waiting for the response
+  // Display the selected value in the view component while waiting, and only switch it back to the
+  // previous value if the response fails
+  const [optimisticDisplayField, setOptimisticDisplayField] = useState<CaseCustomFieldList | null>(
```

Comment:
Nice name 😄  Does it take lot of time to get response here?

Reply:
It takes a couple seconds, not long but long enough to bother me personally 🤣